File: dsa/array/118_pascals_triangle.java
import java.util.*;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>(numRows);

        for (int i = 0; i < numRows; i++){
            result.add(new ArrayList<>(i+1));
            for (int j = 0; j < i + 1; j ++) {
                if ( j == 0 || j == i || i == 0) {
                    result.get(i).add(1);
                }
                else{
                    result.get(i).add(result.get(i-1).get(j-1) + result.get(i-1).get(j));
                }
            }
        }
        return result;
    }
}

File: dsa/array/118_Pascals_Triangle.py
from typing import List
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result: List[List[int]] = []
        for i in range (0, numRows, 1):
            result.append([])
            for j in range(0, i+1, 1):
                if j == 0 or j == i:
                    result[i].append(1)
                else:
                    if i > 1:
                        prevLeftVal: int = result[i-1][j-1] if j >= 0 else 0
                        prevRightVal: int = result[i-1][j]
                        result[i].append(prevLeftVal + prevRightVal)
        return result